base/essentials.jl

```diff
@@ -905,6 +906,12 @@ macro goto(name::Symbol)
 end
 
 # linear indexing
+function getindex(A::GenericMemory, i::Int)
+    @_noub_if_noinbounds_meta
+    @boundscheck ult_int(bitcast(UInt, sub_int(i, 1)), bitcast(UInt, A.length)) || throw_boundserror(A, (i,))
```

Comment:
can we just make this the bootstrap method too? The boundscheck macro is just expanded simply to `@_boundscheck &&`, so this seems compatible

Reply:
sounds reasonable